docs/src/DevelopersGuide.md

```diff
@@ -0,0 +1,142 @@
+# Developer's Guide
+
+This doc is meant to guide any new developers with organization/formatting and common bugs associated with CloudMicrophysics.jl.
+
+## GitHub Pull Requests (PR)
+
+After creating a new branch, create a PR on GitHub.
+    There, you can comment the purpose of your branch and what needs to
+    be done to reach that goal. You can request reviewers to look over
+    and approve your work. 
+
+When creating PR's on GitHub, you will find that every
+    commit & push you make will go through checks. These checks are required
+    to pass before merging.
+The ``JuliaFormatter / format (pull_request)`` check
+    ensures consistent formatting. If this check fails, you can click on details
+    to see where and how your formatting needs to be fixed. If the correction
+    GitHub gives you looks the same as what you already have, it's likely that
+    you have an extra space at the end of the line.
+The ``documenter/deply`` check
+    can be useful when documentation is able to successfully build. Clicking on
+    details will bring up documentation of `CloudMicrophysics.jl` where everything
+    is processed neatly, including example boxes with plots. ``documenter/deploy``
+    and ``Documentation / docbuild (pull_request) `` focuses on documentation only.
+    Any errors will be related to documentation issues such as missing docstrings.
+The ``CodeCov`` and ``ci / ci 1.8.1 `` checks look at CloudMicrophysics library
+    as a whole. Oftentimes, if your files run fine locally, issues are related to
+    tests. See the `Tests` section below for debugging tips.
```

Comment:
```suggestion
- The `codecov` checks what percentage of source code lines are exercised inside the tests.
   We strive to keep the test coverage high, but this test is not strictly required to pass before merging a PR.
- Finally, `ci / ci 1.8.1 ` will run the tests on ubuntu, windows and osx machines. 
   It may happen that the tests pass on your local machine, but fail on one of the machines provided in the cloud for the CI. See the `Tests` section below for debugging tips.
```